### PR TITLE
chore(profiling): add tests for wrapt disable extensions

### PR DESCRIPTION
**Pull request by taegyunkim**
## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by taegyunkim** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9688?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9688** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9688?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9672** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9672?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9687** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9687?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @taegyunkim and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tests/profiling/collector/test_threading.py [hunk 1]
```diff
@@ -537,6 +537,45 @@ def test_anonymous_lock():
     assert release_event.lock_name == "test_threading.py:%d" % release_lineno
 
 
+@pytest.mark.skipif(not os.getenv("WRAPT_DISABLE_EXTENSIONS"), reason="wrapt C extension is disabled")
```
Comment (erikayasuda) [inline tests/profiling/collector/test_threading.py:540]: nit: Maybe these tests can be parameterized? Something like:
```python
@pytest.mark.parametrize(
   "wrapt_disable, init_lock",
   [
      (False, "545"),
      (True, "565")
)
def test_wrapt_c_ext_config(wrapt_disable, init_lock):
...
```

**Review by erikayasuda — APPROVED** [review]
Just one NB, otherwise LGTM

File: tests/profiling/collector/test_threading.py (hunk 1 above)
Comment (taegyunkim) [reply tests/profiling/collector/test_threading.py]: Thanks for pointing that out. I just wanted to test that the results are the same regardless of the WRAPT_DISABLE_EXTENSIONS ENV var, and to be more precise regardless of `_lock.WRAPT_C_EXT` variable. So combined these two tests.

**Pull request merged**